python/pydantic_core/core_schema.py

```diff
@@ -3372,11 +3372,15 @@ def arguments_parameter(
     return _dict_not_none(name=name, schema=schema, mode=mode, alias=alias)
 
 
+VarKwargsMode: TypeAlias = Literal['single', 'unpacked-typed-dict']
+
+
 class ArgumentsSchema(TypedDict, total=False):
     type: Required[Literal['arguments']]
     arguments_schema: Required[List[ArgumentsParameter]]
     populate_by_name: bool
     var_args_schema: CoreSchema
+    var_kwargs_mode: VarKwargsMode
```

Comment:
As discussed with DH, we decided to have everything coupled together in the `ArgumentsSchema`, as having a single `VarKwargsSchema` did not really make sense and made things harder to handle in pydantic core.

This has the downside that `var_kwargs_schema` serves two purposes, depending on `var_kwargs_mode`: a core schema to validate every value of each extra kwarg, or a `TypedDictSchema` to validate `kwargs` entirely. Maybe we could make them two separate keys?

Reply:
> WDYM by this?

if `var_kwargs_mode` is `'single'`, then `var_kwargs_single_schema` would have to be provided. Otherwise if `var_kwargs_mode` is `'unpacked-typed-dict`, `var_kwargs_unpacked_typed_dict_schema` would have to be provided. This makes things more explicit but with more validation logic involved so probably best to keep it as is.

Reply:
Agreed.